presto-main/src/main/java/com/facebook/presto/catalogserver/RemoteMetadataManager.java

```diff
@@ -24,30 +24,54 @@
 import com.facebook.presto.metadata.ViewDefinition;
 import com.facebook.presto.spi.ConnectorMaterializedViewDefinition;
 import com.facebook.presto.spi.TableHandle;
+import com.facebook.presto.transaction.TransactionInfo;
 import com.facebook.presto.transaction.TransactionManager;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 
 import javax.inject.Inject;
 
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Optional;
+import java.util.OptionalLong;
 
+import static com.facebook.presto.common.RuntimeUnit.NONE;
 import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
 // TODO : Use thrift to serialize metadata objects instead of json serde on catalog server in the future
 // TODO : Add e2e tests for this class
 public class RemoteMetadataManager
         extends DelegatingMetadataManager
 {
+    private static final String CATALOG_SERVER_CACHE_HIT_COUNT = "CATALOG_SERVER_CACHE_HIT_COUNT";
+    // TODO make cache constants configurable
+    private static final OptionalLong cacheExpiresAfterWriteMillis = OptionalLong.of(600000);
+    private static final long cacheMaximumSize = 1;
```

Comment:
Let's reuse values defined in `CatalogServer` instead of creating its own copies.

Reply:
Makes sense.